corehq/apps/hqadmin/management/commands/force_web_user_password_reset.py

```diff
@@ -0,0 +1,90 @@
+
+    Usage:
+        $ ./manage.py force_web_user_password_reset web_users_file
+    """
+
+    def add_arguments(self, parser):
+        parser.add_argument('web_users_file')
+        parser.add_argument('--skip-if-reset-since', type=utc_timestamp,
+                            help="Timestamp of the last_password_set after which we will not force a reset."
+                                 " Format as YYYY-MM-DD HH:MM:SS")
+
+    def handle(self, web_users_file, skip_if_reset_since, **options):
+        web_user_usernames = get_lines_from_file(web_users_file)
+        errors = []
+        web_users_to_reset = []
+        web_users_to_skip_due_to_recent_reset = []
+        for web_user_username in web_user_usernames:
+            web_user = WebUser.get_by_username(web_user_username)
+            if web_user is None:
+                errors.append(f'No user named {web_user_username}')
+            elif not web_user.is_web_user():
+                errors.append(f'{web_user_username} is not a WebUser')
+            elif skip_if_reset_since and web_user.last_password_set \
+                    and web_user.last_password_set > skip_if_reset_since:
+                web_users_to_skip_due_to_recent_reset.append(web_user)
+            else:
+                web_users_to_reset.append(web_user)
+
+        if errors:
+            raise CommandError(f'The following errors were found in your input file:\n{errors}')
+
+        if web_users_to_skip_due_to_recent_reset:
+            self.stdout.write("The following users will be skipped "
+                              "because they have reset their passwords since {skip_if_reset_since}")
+
+            for web_user in web_users_to_skip_due_to_recent_reset:
+                self.stdout.write(f"  - {web_user.username} (password last set on {web_user.last_password_set})")
+
+        if not web_users_to_reset:
+            print("No web users to reset passwords for. No action taken.")
+            return
+
+        self.stdout.write("The following users will be logged out and have their passwords force reset.")
+        self.stdout.write("Additionally, they will receive an email directing them to reset their password.")
+        for web_user in web_users_to_reset:
+            self.stdout.write(f"  - {web_user.username}")
+
+        if 'y' != input('Do you want to proceed? [y/N]'):
+            raise CommandError('You have aborted the command and no action was taken.')
+
+        for web_user in web_users_to_reset:
+            self.stdout.write(f"Force resetting password for {web_user.username}", ending='')
+            force_password_reset(web_user)
+            self.stdout.write(" - Done")
+
+
+def get_lines_from_file(filename):
+    with open(filename) as f:
+        return [line.strip() for line in f.readlines()]
+
+
+def force_password_reset(web_user):
+    user = web_user.get_django_user()
+    user.set_password(uuid.uuid4().hex)
```

Comment:
Will this update the `last_password_set` field on the couch user? That appears to be a couch-specific field, but we're only updating a django model here.

Reply:
I'm intentionally not updating that field, because I'm treating this operation not as us "resetting their password" but as us "unsetting" their password to force them to reset it. Until they actually reset it, I want that field to read the date of the last time their password was set (to something the client/user controls).